pkg/controller/dnsendpoint/nameserver/azure.go

```diff
@@ -119,21 +121,26 @@ func (q *azureQuery) recordSet(domain string, values sets.String) dns.RecordSet
 
 // queryNameServer queries Azure for the name servers for the specified domain in the specified managed zone.
 func (q *azureQuery) queryNameServer(azureClient azureclient.Client, rootDomain string, domain string) (sets.String, error) {
-	recordSets, err := azureClient.ListRecordSetsByZone(context.TODO(), q.resourceGroupName, rootDomain)
+	recordSetsPage, err := azureClient.ListRecordSetsByZone(context.TODO(), q.resourceGroupName, rootDomain, to.Int32Ptr(50))
 	if err != nil {
 		return nil, err
 	}
 
-	for _, recordSet := range *recordSets {
-		if *recordSet.Type != string(dns.NS) {
-			continue
-		}
-		if *recordSet.Name == domain {
-			values := sets.NewString()
-			for _, v := range *recordSet.NsRecords {
-				values.Insert(*v.Nsdname)
+	for recordSetsPage.NotDone() {
+		for _, recordSet := range recordSetsPage.Values() {
+			if *recordSet.Type != string(dns.NS) {
+				continue
+			}
+			if *recordSet.Name == domain {
```

Comment:
Can we use the `recordsetnamesuffix` parameter to `ListByDNSZone` to filter on the server side rather than the client side?

Reply:
We can. But won't we still have to loop through? Or we'd expect a slice with a length of 1

Reply:
You would still have to loop through. The point is not to keep from needing to loop. The point is to reduce the traffic from the server when we can easily.

Reply:
agreed. I will use recordsetnamesuffix then